### ISSUE TITLE
Version number is missing from dotnet publish output in alpine

### ISSUE THREAD

**Issue by npuc85** [dotnet/sdk#12913]
In my CI pipeline I am using an alpine image to build, test, and publish my project as a single file. This is the command I am using to publish:
`dotnet publish -c Release -r win-x64 /p:PublishReadyToRun=false /p:PublishSingleFile=true /p:PublishTrimmed=true`
However, this is creating an exe that has no version number. I do have a Version specified inside of my .csproj file, and I have tried both with and without using the `/p:Version=1.2.3` argument on the `dotnet publish` command. Both ways result in an exe file that has no version number.

Running the same command locally on my Windows machine does produce an exe that has a version number.
Also, `dotnet build` (both in Windows and alpine) produces an exe that has a version number. It seems to only be an issue with the `publish` command and only in the alpine image.

**Comment by clane2812** [comment]
yes - can confirm that this is still the case with the net5 sdk in the image 'FROM mcr.microsoft.com/dotnet/runtime-deps:5.0'. The version is set for the dll on normal publish, but not for the exe when you use /p:PublishSingleFile=true
Publishing on windows works fine, the version is set there in the exe as well.

**Comment by clane2812** [comment]
This topic is discussed and yet open 
https://github.com/dotnet/sdk/issues/4127

**Comment by PallaviThanvi** [comment]
Any updates on this?

**Comment by CodyPaul** [comment]
Also experiencing this issue on Ubuntu based Azure Pipeline agents when targeting dotnet core builds for win-x64

**Comment by GOATS2K** [comment]
I also have this issue!

I used the following command to build my project:

`dotnet publish -c Release -r ${{ matrix.platform }} --self-contained -o ${{ matrix.folder_name }} -p:SkipSwaggerGen=true -p:EnableCompressionInSingleFile=true -p:FileVersion=version -p:Version=version -p:AssemblyFileVersion=version`

**Comment by eric10k93** [comment]
Just an update on this, in case it's helpful to anyone.  I realized that for a .exe file build, dotnet ALSO produces a .dll file.  I WAS building with SingleFilePublish=true, but setting this to false allows me to see the .dll file that dotnet produces.  The FileVersionInfo is then contained within the .dll file, just not the .exe
